### ISSUE TITLE
Improve the Stripe set up task flow

### ISSUE THREAD

**Issue by pmcpinto** [woocommerce/woocommerce-admin#4431]
Without the possibility of allowing users to automatically create a Stripe account during the store setup, it becomes less intuitive in the UI that they need to create an account: https://d.pr/i/4SaVQ3
Probably this is also influencing the drop in terms of Stripe accounts created.

A good workaround could be adding more emphasis to the Create account CTA that exists in this step.

**Comment by pmcpinto** [comment]
@jameskoster what do you think it should be the best option here?

**Comment by jameskoster** [comment]
Without doing some extra work to detect if the user has an account / is logged in already, I think all we can really do is draw more attention to where the user should collect this information, via the [Notice](https://github.com/WordPress/gutenberg/tree/wp/5.4/packages/components/src/notice) component:

![notice](https://cldup.com/KT5D2MoHfl-3000x3000.png)

**Comment by pmcpinto** [comment]
@jameskoster that could work, but perhaps it would be more effective if it's displayed before the input fields. What do you think?

**Comment by jameskoster** [comment]
I don't think it will make too much difference, as long as the notice is inside the viewport. Moving it above the inputs increases the likelihood of that being the case I suppose - so sure - we can move it :) 

I suppose to make doubly sure folks know where to go, we could add links to the input helper text too:

![account](https://cldup.com/pshv75v4Ac-3000x3000.png)

**Comment by pmcpinto** [comment]
@jameskoster should we test a remind me later option that triggers an inbox notification? The Stripe and WCPay account creation requires a lot of info, so probably there's a significant % of users that don't want to deal with that at that moment, need to spend more time looking at the pricing, etc.

**Comment by jameskoster** [comment]
@pmcpinto I think that's a good idea, but we should make the UX invisible so that folks aren't _encouraged_ to do it later.

The back button in the app bar already provides an affordance for people to exit setup (and indicate they want to do this later).

If someone installs the plugin but doesn't connect an account in x days, we should send the Inbox reminder.

**Comment by thenbrent** [comment]
Rather than requiring API creds, it looks like it's possible to display a _Connect_ button or similar that allows merchants to connect a Stripe account, either existing or new, from within the on-boarding. Even when Jetpack isn't installed or connected. For more information, see pb0GrA-A1-p2 #comment-798

**Comment by pmcpinto** [comment]
> Rather than requiring API creds, it looks like it's possible to display a Connect button or similar that allows merchants to connect a Stripe account, either existing or new, from within the on-boarding.

That would be great. @timmyc I'm also going to add this issue to the Cycle 4 scoping, for consideration

**Comment by jeffstieler** [comment]
@thenbrent assuming the API endpoint work will be done before we start this, is it as simple as creating a button that links to a URL?

**Comment by mattsherman** [comment]
I don't know enough about how Stripe integration works to give an estimate, but if it boils down to adding a butter routing to a specific URL, I'd say 1.

**Comment by becdetat** [comment]
It isn't clear to me what has been decided for this ticket. Are we adding a notice component? An inbox note? Or are we instead adding a Connect button? cc @pmcpinto

**Comment by pmcpinto** [comment]
> It isn't clear to me what has been decided for this ticket. Are we adding a notice component? An inbox note? Or are we instead adding a Connect button? cc @pmcpinto

Adding the Connect button seems to be the option that can have the biggest impact. Assuming that this is a low effort change. @thenbrent is there anyone on the Hydra side that can provide more info about adding the Connect btn?

**Comment by becdetat** [comment]
Hi @thenbrent, just wondering if you have any more info on this? Thanks :-)

**Comment by dechov** [comment]
cc @chickenn00dle in case you have any estimate on when the connection flow [mentioned above](https://github.com/woocommerce/woocommerce-admin/issues/4431#issuecomment-645763505) could be ready for integration with the Stripe onboarding task.

**Comment by chickenn00dle** [comment]
Hey @dechov 

The Connect Server changes for this have just been approved and I'm assuming will be merged shortly: 1443-gh-woocommerce-connect-server

Per pb0GrA-Pd-p2, the next step is for the Stripe PR to be reviewed and merged: https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1267 

This would probably be the largest time investment before everything is all ready to go as the WC-Admin PR for this is not too big: https://github.com/woocommerce/woocommerce-admin/pull/4933

In terms of estimates, I guess as long as it would take for someone in Hydra to review the Stripe PR. On my end, I'm happy to focus my efforts on responding to any feedback ASAP to get this done.

**Comment by chickenn00dle** [comment]
> is that what we want, or should we still offer the key input as an option?

This was the default behavior when we removed default account activation: https://github.com/woocommerce/woocommerce-admin/pull/4164

A merchant would attempt to auto-create an account via email and manual inputs would only become available upon failure I believe. So when defaulting to OAuth, I had assumed this was the expected behavior as well.

If for some reason we wanted to change this, I imagine that wouldn't be too hard to do though.

**Comment by dechov** [comment]
Personally, I think a) that we should still provide a (secondary, in this case) option to enter keys manually, and b) that the notice as designed above is a good enhancement independent of the OAuth route.

I was also thinking we can link "Stripe account" directly to https://dashboard.stripe.com/apikeys – and maybe have the doc link in a "more information" parenthetical?

**Comment by chickenn00dle** [comment]
> a) that we should still provide a (secondary, in this case) option to enter keys manually

I've addressed this here: https://github.com/woocommerce/woocommerce-admin/pull/4933#issuecomment-704051344

**Comment by samueljseay** [comment]
It's really hard to follow this thread and determine what is left to be done. I'd appreciate if in future we could update the requirements in the description to match conversation outcomes.

I'm still not clear on what is left to be done here, but let me try summarise. It looks like #4933 partially improves the flow, here is my assumption of what is still left based on the discussion here:

1. Implement the connect button so by default a user would not enter any credentials for Stripe?
2. If the oauth connection fails then when the user returns the input fields are available?
3. Do we also provide the notice that was designed in the case of failure?

cc @dechov @chickenn00dle @pmcpinto 

Also side note: Assuming its all 3 of these things I think it's a bit more effort than was originally estimated.

**Comment by samueljseay** [comment]
bump on this @dechov @chickenn00dle @pmcpinto

**Comment by chickenn00dle** [comment]
Hey @samueljseay 👋 

It's my understanding steps 1 and 2 were complete in https://github.com/woocommerce/woocommerce-admin/pull/4164 

At least when setting up Stripe during the payment task of the task list, merchants should first see a connect button to connect via OAuth. And if that fails for whatever reason, they can then input credentials manually.

The above was actually done before #4933. In #4933 we only removed the need for Jetpack and WC Services to be installed in order for the Stripe connect button to work.

**Comment by samueljseay** [comment]
Thanks so much @chickenn00dle that clarifies a lot. 

So it sounds like we are just implementing an emphasis notice as per the designs here. Sound right @pmcpinto ?

**Comment by pmcpinto** [comment]
> So it sounds like we are just implementing an emphasis notice as per the designs here. Sound right @pmcpinto ?

Yes. That's correct.

**Comment by louwie17** [comment]
I was just looking at this ticket briefly and noticed that the info Notice, [has turned white](https://wordpress.github.io/gutenberg/?path=/story/components-notice--default) as you can see in that link, and here:
<img width="648" alt="Screen Shot 2021-01-19 at 1 52 12 PM" src="https://user-images.githubusercontent.com/2240960/105073711-c9db8f80-5a5d-11eb-8869-7b41eef9c341.png">

@pmcpinto as part of the design do we want to keep it a light blue as displayed in the designs here: https://github.com/woocommerce/woocommerce-admin/issues/4431#issuecomment-635217037 or stick with the white?

**Comment by pmcpinto** [comment]
I'm good with sticking with the white. @elizaan36 what do you think?

**Comment by elizaan36** [comment]
This notification got lost! The white background for the banner is fine, it's one of the core WP notice styles👍
